source/server/http/admin.cc

```diff
@@ -376,36 +376,18 @@ void AdminImpl::writeClustersAsJson(Buffer::Instance& response) {
         Network::Utility::addressToProtobufAddress(*host->address(),
                                                    *host_status.mutable_address());
         host_status.set_hostname(host->hostname());
-        std::vector<Stats::CounterSharedPtr> sorted_counters;
-        for (const Stats::CounterSharedPtr& counter : host->counters()) {
-          sorted_counters.push_back(counter);
-        }
-        std::sort(
-            sorted_counters.begin(), sorted_counters.end(),
-            [](const Stats::CounterSharedPtr& counter1, const Stats::CounterSharedPtr& counter2) {
-              return counter1->name() < counter2->name();
-            });
 
-        for (const Stats::CounterSharedPtr& counter : sorted_counters) {
+        for (const auto& named_counter : host->counters()) {
           auto& metric = *host_status.add_stats();
-          metric.set_name(counter->name());
-          metric.set_value(counter->value());
+          metric.set_name(named_counter.first);
+          metric.set_value(named_counter.second.get().value());
           metric.set_type(envoy::admin::v2alpha::SimpleMetric::COUNTER);
         }
 
-        std::vector<Stats::GaugeSharedPtr> sorted_gauges;
-        for (const Stats::GaugeSharedPtr& gauge : host->gauges()) {
-          sorted_gauges.push_back(gauge);
-        }
-        std::sort(sorted_gauges.begin(), sorted_gauges.end(),
-                  [](const Stats::GaugeSharedPtr& gauge1, const Stats::GaugeSharedPtr& gauge2) {
-                    return gauge1->name() < gauge2->name();
-                  });
-
-        for (const Stats::GaugeSharedPtr& gauge : sorted_gauges) {
+        for (const auto& named_gauge : host->gauges()) {
           auto& metric = *host_status.add_stats();
-          metric.set_name(gauge->name());
-          metric.set_value(gauge->value());
+          metric.set_name(named_gauge.first);
```

Comment:
I'm not sure if this matters (or if I'm right) but this looks to me like it won't give the same answer. I think the old system might have yielded a scoped name here, and the new system looks like it won't include the scope. Can you check?

Reply:
Hosts were using isolated stats store.  I think a scope name was never passed in.  There are some tests for the output of admin.cc in //test/server/http:admin_test they don't provide e2e coverage against a real HostStat object.  If scope name was supported for host stats, there are no tests that cover it.